src/storage/src/hummock/sstable/block_iterator.rs

```diff
@@ -72,7 +73,11 @@ impl BlockIterator {
 
     pub fn key(&self) -> FullKey<&[u8]> {
         assert!(self.is_valid());
-        FullKey::decode(&self.key)
+        let table_id = TableId::new(self.block.table_id());
+        let epoch_pos = self.key[..].len() - EPOCH_LEN;
+        let epoch = (&self.key[epoch_pos..]).get_u64();
+        let user_key = UserKey::new(table_id, TableKey(&self.key[..epoch_pos]));
+        FullKey::from_user_key(user_key, epoch)
```

Comment:
Can we maintain `self.key` as `FullKey<Vec<u8>>` instead of `BytesMut`? There are severl benefits:
1.We can just return self.key directly in `pub fn key()`
2. We can avoid memory allocation and copy for key encoding for the seek key in the `seek` and `seek_le`

Reply:
Here `self.key` is `table_key|epoch`, which does not contain `table_id`, if change to `fullkey` struct, we need to construct `fullkey` via `table_id` every time `seek_restart_point_by_index`  and `next`.

Reply:
How about using a separate structure to maintain table_key|epoch, and only splice the table_id when it is returned?

Reply:
I will try to avoid encode `fullkey`, but compare unencoded `fullkey` struct to  improve performance.